src/inc/gcinfotypes.h

```diff
@@ -310,12 +310,13 @@ inline ReturnKind GetStructReturnKind(ReturnKind reg0, ReturnKind reg1)
     return structReturnKind;
 }
 
-inline ReturnKind ExtractRegReturnKind(ReturnKind returnKind, size_t regNo)
+inline ReturnKind ExtractRegReturnKind(ReturnKind returnKind, size_t regNo, bool& moreRegs)
 {
     _ASSERTE(IsValidReturnKind(returnKind));
     _ASSERTE(IsValidReturnRegister(regNo));
 
     ReturnKind regReturnKind = (ReturnKind)((returnKind >> (regNo * 2)) & 3);
+    moreRegs = (returnKind >> ((regNo+1) * 2)) != 0;
```

Comment:
A comment explaining the voodoo that is used to compute regReturnKind and set moreRegs would seem essential here.